katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,48 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  return unless facts[:os][:release][:major] == '7'
+
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
```

Comment:
I think we should move the disk space check to `katello/hooks/pre_validations` so we don't even start an upgrade unless there's sufficient disk space. Failing an upgrade leaves a feeling of an inconsistent state if it's not really.